File: app/src/main/java/com/cliniva/enventory/ui/transaction/UnpaidFragment.java
package com.cliniva.enventory.ui.transaction;


import android.databinding.DataBindingUtil;
import android.databinding.ViewDataBinding;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.design.widget.FloatingActionButton;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.cliniva.enventory.R;
import com.cliniva.enventory.adapter.RecyclerViewAdapter;
import com.cliniva.enventory.adapter.base.BaseRecyclerClickListener;
import com.cliniva.enventory.adapter.base.BaseRecyclerViewHolder;
import com.cliniva.enventory.model.Transaction;
import com.cliniva.enventory.ui.add.AddActivity;
import com.cliniva.enventory.ui.base.BaseFragment;
import com.cliniva.enventory.utils.IntentUtils;
import com.cliniva.enventory.viewholder.UnpaidHolder;

import java.util.List;

/**
 * A simple {@link Fragment} subclass.
 */
public class UnpaidFragment extends BaseFragment implements TransactionContract.View, View.OnClickListener, BaseRecyclerClickListener<Transaction> {

    private TransactionContract.Presenter mTransactionPresenter;
    private RecyclerView mRecyclerView;
    private static final String TAG = "UnpaidFragment";

    public UnpaidFragment() {
        // Required empty public constructor
    }

    public static UnpaidFragment getInstance(){
        UnpaidFragment fragment = new UnpaidFragment();
        Bundle args = new Bundle();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mTransactionPresenter = new TransactionPresenter(this);
    }

    @Override
    public int getLayoutRes() {
        return R.layout.fragment_unpaid;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        mRecyclerView = view.findViewById(R.id.rv_unpaid_view);
        LinearLayoutManager layoutManager = new LinearLayoutManager(mTransactionPresenter.getContext());
        mRecyclerView.setLayoutManager(layoutManager);
        mTransactionPresenter.onLoad();
        Log.d(TAG, "onViewCreated: ");

        FloatingActionButton fab = view.findViewById(R.id.btn_fab);
        fab.setOnClickListener(this);
    }

    @Override
    public void setList(List<Transaction> unpaidList) {
        RecyclerViewAdapter<Transaction, BaseRecyclerClickListener<Transaction>> unpaidAdapter = new RecyclerViewAdapter<Transaction, BaseRecyclerClickListener<Transaction>>(unpaidList) {
            @NonNull
            @Override
            public BaseRecyclerViewHolder<Transaction, BaseRecyclerClickListener<Transaction>> onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
                ViewDataBinding binding = DataBindingUtil.inflate(LayoutInflater.from(mTransactionPresenter.getContext()), R.layout.item_unpaid, parent, false);
                return new UnpaidHolder(binding);
            }
        };
        unpaidAdapter.setListener(this);
    }

    @Override
    public void showToast(String message) {
        Toast.makeText(mTransactionPresenter.getContext(), message, Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()){
            case R.id.btn_fab:
                IntentUtils.getInstance()
                        .onActivityIntentWithoutExtras(getActivity(), AddActivity.class);
                break;
        }
    }

    @Override
    public void onItemClickListener(Transaction item, int position) {

    }
}

File: app/src/main/java/com/amirsons/inventory/ui/fragment/transaction/UnpaidFragment.kt
package com.amirsons.inventory.ui.fragment.transaction


import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.recyclerview.widget.LinearLayoutManager
import com.amirsons.inventory.R
import com.amirsons.inventory.ui.recyclerview.base.RecyclerViewAdapter
import com.amirsons.inventory.ui.recyclerview.base.BaseRecyclerClickListener
import com.amirsons.inventory.ui.recyclerview.base.BaseRecyclerViewHolder
import com.amirsons.inventory.datamanager.model.Transaction
import com.amirsons.inventory.ui.activity.transaction.TransactionActivity
import com.amirsons.inventory.ui.base.BaseFragment
import com.amirsons.inventory.utils.IntentUtils
import com.amirsons.inventory.ui.recyclerview.viewholder.UnpaidHolder
import kotlinx.android.synthetic.main.fragment_unpaid.*
import java.util.*

/**
 * A simple [BaseFragment] subclass.
 */
class UnpaidFragment : BaseFragment(), UnpaidView, View.OnClickListener {

    private lateinit var mTransactionPresenter: UnpaidPresenter

    override val contentLayout: Int
        get() = R.layout.fragment_unpaid

    companion object {
        val instance: UnpaidFragment
            get() {
                val fragment = UnpaidFragment()
                val args = Bundle()
                fragment.arguments = args
                return fragment
            }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mTransactionPresenter = UnpaidMvp(this)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val layoutManager = LinearLayoutManager(mTransactionPresenter.context)
        rv_unpaid_view.layoutManager = layoutManager
        
        mTransactionPresenter.onLoad()

        btn_fab.setOnClickListener(this)
    }

    override fun setList(dataList: ArrayList<Transaction>) {

        val unpaidAdapter = object : RecyclerViewAdapter<Transaction, BaseRecyclerClickListener<Transaction>>(dataList) {
            override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseRecyclerViewHolder<Transaction, BaseRecyclerClickListener<Transaction>> {
                return UnpaidHolder(inflate(R.layout.item_transaction_history, parent))
            }
        }

        unpaidAdapter.setListener(object : BaseRecyclerClickListener<Transaction>{

            override fun onItemClickListener(item: Transaction, position: Int) {
                TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
            }
        })

        rv_unpaid_view.adapter = unpaidAdapter
    }

    override fun showToast(message: String) {
        Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.btn_fab -> IntentUtils.instance
                    .onActivityIntentWithoutExtras(context, TransactionActivity::class.java)
        }
    }
}
